hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -1867,10 +1869,15 @@ public String selectFragment(String alias, String suffix) {
 								FetchTiming fetchTiming = fetchable.getMappedFetchOptions().getTiming();
 								final boolean selectable;
 								if ( fetchable instanceof AttributeMapping ) {
-									final int propertyNumber = ( (AttributeMapping) fetchable ).getStateArrayPosition();
-									final int tableNumber = getSubclassPropertyTableNumber( propertyNumber );
-									selectable = !isSubclassTableSequentialSelect( tableNumber )
-											&& propertySelectable[propertyNumber];
+									if ( fetchParent instanceof EmbeddableFetchImpl && ( (EmbeddableFetchImpl) fetchParent ).getReferencedMappingContainer() instanceof EntityIdentifierMapping ) {
```

Comment:
How about this instead?

```suggestion
									if ( fetchParent instanceof EmbeddableResultGraphNode && ( (EmbeddableResultGraphNode) fetchParent ).getReferencedMappingContainer() instanceof EntityIdentifierMapping ) {
```

Reply:
@beikov,  I changed it to `if ( fetchParent instanceof EmbeddableResultGraphNode && ( (EmbeddableResultGraphNode) fetchParent ).getReferencedMappingContainer() == getIdentifierMapping() )`, what do you think?

Reply:
Even better :)